client/src/store/utils/thunkCreators.js

```diff
@@ -91,6 +92,35 @@ const sendMessage = (data, body) => {
   });
 };
 
+const transmitConversationRead = (conversationId) => {
+  socket.emit("conversation-read", conversationId);
+};
+
+// attach userId to the new message before it gets to the reducer util function
+// and check if the message belongs to the current active conversation
+export const receiveNewMessage = (message, sender) => (dispatch, getGlobalState) => {
+  const { activeConversation, user, conversations } = getGlobalState();
+  const activeConvo = conversations.find(convo => convo.otherUser.username === activeConversation);
+  const activeConvoId = activeConvo && activeConvo.id;
```

Comment:
Small nitpick for more modern syntax:

```suggestion
  const activeConvoId = activeConvo?.id;
```